src/main/java/org/prebid/server/bidder/seedingAlliance/SeedingAllianceBidder.java

```diff
@@ -42,26 +53,52 @@ public SeedingAllianceBidder(String endpointUrl, JacksonMapper mapper) {
 
     @Override
     public final Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
-        final BidRequest resolvedRequest = resolveRequest(bidRequest);
+        String seatId = null;
+        final List<Imp> modifiedImps = new ArrayList<>();
+        for (Imp imp: bidRequest.getImp()) {
+            try {
+                final ExtImpSeedingAlliance impExt = parseImpExt(imp);
+                seatId = impExt.getSeatId();
+                final Imp modifiedImp = imp.toBuilder().tagid(impExt.getAdUnitId()).build();
+                modifiedImps.add(modifiedImp);
+            } catch (PreBidException e) {
+                return Result.withError(BidderError.badInput(e.getMessage()));
+            }
+        }
+        final BidRequest modifiedBidRequest = modifyBidRequest(bidRequest, modifiedImps);
 
         return Result.withValue(
                 HttpRequest.<BidRequest>builder()
                         .method(HttpMethod.POST)
-                        .uri(endpointUrl)
+                        .uri(makeEndpoint(seatId))
                         .headers(HttpUtil.headers())
-                        .body(mapper.encodeToBytes(resolvedRequest))
-                        .impIds(BidderUtil.impIds(resolvedRequest))
-                        .payload(resolvedRequest)
+                        .body(mapper.encodeToBytes(modifiedBidRequest))
+                        .impIds(BidderUtil.impIds(modifiedBidRequest))
+                        .payload(modifiedBidRequest)
                         .build());
     }
 
-    private static BidRequest resolveRequest(BidRequest bidRequest) {
+    private String makeEndpoint(String seatId) {
+        final String accountId = Optional.ofNullable(seatId).filter(StringUtils::isNotBlank).orElse(DEFAULT_SEAT_ID);
```

Comment:
No need to use `Optional` here:
`StringUtils.isNotBlank(seatId) ? seatId : DEFAULT_SEAT_ID;`